### DISCUSSION TITLE
Smoke gets cut off when using screen scaling

### DISCUSSION THREAD

**Discussion by Daste745** [ppy/osu#20897] [Q&A]
### Type

Cosmetic

### Bug description

When drawing with the smoke cursor with screen scaling enabled (set to 80%/80% on the attached screenshot), the smoke gets cut off after reaching the original screen boundaries.

The expected result is being able to draw on the whole screen.

### Screenshots or videos

![image](https://user-images.githubusercontent.com/25880357/197415739-81caebca-53ad-4b98-8cea-db59c1aa5e6e.png)

### Version

2022.1022.0-lazer

### Logs

[performance.log](https://github.com/ppy/osu/files/9847441/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9847442/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9847443/updater.log)
[database.log](https://github.com/ppy/osu/files/9847444/database.log)
[input.log](https://github.com/ppy/osu/files/9847445/input.log)
[network.log](https://github.com/ppy/osu/files/9847446/network.log)

**Comment by peppy** [comment]
I don’t foresee this changing. If anything, the “gameplay” scaling will be removed as an option and replaced with a different system, but do keep in mind that smoke is a gameplay level element and is saved to a the replay. Drawing outside the playfield doesn’t make much sense.

**Reply by Daste745** [reply]
Yeah, I see. The "gameplay" scaling is useful for me, as I use a large screen and osu! gameplay on 100% scaling is uncomfortable, but the UI is fine. For me, the smoke could be drawable outside the play area, but not saved in the replay. Maybe it could be indicated by some sort of border around the scaled screen (like for the play area), that anything outside of it will not be visible on the replay? The sudden smoke cutoff is a bit unnatural.
